File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int prod = 1;
        for(int i=0; i<nums.length; i++) {
            if(nums[i] < 0)
                prod *= -1;
            else if(nums[i] > 0)
                prod *= 1;
            else
                return 0;
        }
        return prod;
    }
    // public int signFunc(int t) {
    //     if(t < 0)
    //         return -1;
    //     if(t > 0)
    //         return 1;
    //     else
    //         return 0;
    //     //return (t != 0 && t > 0) ? 1 : 0;
    // }
}

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.cs
public class Solution {
    public int ArraySign(int[] nums) {
        //return nums.Any(x => x== 0) ? 0: nums.Count( a => a < 0) % 2 == 1 ? -1 :1;
        int minus = 0;
        foreach(int i in nums) {
            if(i == 0)
                return 0;
            if(i < 0)
                minus++;
        }
        return minus%2 == 0 ? 1 : -1;
    }
}